### PR TITLE
fix(select): popover uses modern form syntax

### LINKED ISSUE #27071: bug: `ion-select` with popover interface shows warnings when opened
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

When we have `<ion-select>` input with `interface="popover"` and we open the popover, the following warnings are thrown for each option:
```
[Ionic Warning]: ion-radio is being used with the "legacy" property enabled which will forcibly enable the legacy form markup. This property will be removed in an upcoming major release of Ionic where this form control will use the modern form markup.

Developers can dismiss this warning by removing their usage of the "legacy" property and using the new radio syntax. 
<ion-radio class="sc-ion-select-popover-ios">
```
This can be observed even on the official components docs: 
https://ionicframework.com/docs/api/select#popover

### Expected Behavior

No errors or warnings to be logged.

### Steps to Reproduce

1. Add `<ion-select>` component with a popover interface (`interface="popover"`) and add a few options inside. You can also just open the official example at 
https://ionicframework.com/docs/api/select#popover
2. Open the browser console
3. Click on the select to open the popover

### Code Reproduction URL

 https://ionicframework.com/docs/api/select#popover

### Ionic Info

Ionic:

   Ionic CLI                     : 6.20.9 (/usr/lib/node_modules/@ionic/cli)
   Ionic Framework               : not installed
   @angular-devkit/build-angular : 15.2.4
   @angular-devkit/schematics    : 15.2.4
   @angular/cli                  : 15.2.4
   @ionic/angular-toolkit        : 9.0.0

Capacitor:

   Capacitor CLI      : 4.7.1
   @capacitor/android : not installed
   @capacitor/core    : 4.7.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.2

System:

   NodeJS : v18.15.0 (/usr/bin/node)
   npm    : 9.6.2
   OS     : Linux 5.19

### Additional Information

_No response_

### LINKED ISSUE #27786: bug: legacy radios in ion-select creates retainer when used in popover
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

I have created a 'ion-select' with interface set as popover.

On selection of an select-option, the DOM Nodes created for displaying the options are not released

### Expected Behavior

DOM Nodes should be released upon selection / dismissal

### Steps to Reproduce

        <ion-select interface="popover" placeholder="testing">
          <ion-select-option *ngFor="let a of [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1], let i = index">
            {{ "page " + (i + 1) }}
          </ion-select-option>
        </ion-select>

the above following code generates the said behaviour. DOM Node count can be monitored in the 'Performance Monitor' tool

### Code Reproduction URL

_No response_

### Ionic Info

Ionic:

   Ionic CLI                     : 7.1.1 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 7.0.2
   @angular-devkit/build-angular : 15.2.6
   @angular-devkit/schematics    : 15.2.8
   @angular/cli                  : 15.2.8
   @ionic/angular-toolkit        : 9.0.0

Capacitor:

   Capacitor CLI      : 4.7.3
   @capacitor/android : not installed
   @capacitor/core    : 4.7.3
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.2

System:

   NodeJS : v18.16.0 (/home/yg/.nvm/versions/node/v18.16.0/bin/node)
   npm    : 9.5.1
   OS     : Linux 5.15

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: resolves #27071, resolves #27786

---------

## What is the current behavior?

`ion-select-popover` is using the legacy syntax for `ion-checkbox` and `ion-radio`.

## What is the new behavior?

- Migrated checkbox and radio to use the modern syntax
- Updated the visual rendering tests to also check the checked state. I noticed during development that I accidentally broke the checked state, so I added tests so we can't accidentally break it again.

A couple notes on other screenshot diffs:

- On iOS, the item border line now extends past the radio circle. This is an intentional behavior and aligns with native iOS. Having the radio in the "start" slot is typically only done when the content in the default slot is another interactive element (like and input) and not a text label.
- On MD, the control heights are smaller. When comparing with https://material-components.github.io/material-components-web-catalog/#/component/select, the new behavior seems to be correct. Both the spec and Ionic item heights are 48px. Interestingly, the radios in `ion-select-popover` have always been 48px tall, but the checkboxes were 54px. Now both the radios and checkboxes are 48px.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/select-popover/test/basic/select-popover.e2e.ts [hunk 1]
```diff
@@ -81,11 +81,11 @@ configs({ directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
       selectPopoverPage = new SelectPopoverPage(page);
     });
     test('should not have visual regressions with single selection', async () => {
-      await selectPopoverPage.setup(config, options, false);
+      await selectPopoverPage.setup(config, checkedOptions, false);
       await selectPopoverPage.screenshot(screenshot, 'select-popover-diff');
     });
     test('should not have visual regressions with multiple selection', async () => {
-      await selectPopoverPage.setup(config, options, true);
+      await selectPopoverPage.setup(config, checkedOptions, true);
```
Comment (liamdebeasi) [inline core/src/components/select-popover/test/basic/select-popover.e2e.ts:88]: I noticed we did not have any screenshots for checked values which is why I switched to `checkedOptions` here

File: core/src/components/select-popover/select-popover.tsx [hunk 2]
```diff
@@ -116,19 +116,28 @@ export class SelectPopover implements ComponentInterface {
 
   renderCheckboxOptions(options: SelectPopoverOption[]) {
     return options.map((option) => (
-      <ion-item class={getClassMap(option.cssClass)}>
+      <ion-item
+        class={{
+          // TODO FW-4784
```
Comment (liamdebeasi) [inline core/src/components/select-popover/select-popover.tsx:121]: In the future we will be able to use `:has` instead, but it's still too new to use in Ionic: https://caniuse.com/?search=%3Ahas

File: core/src/components/select-popover/test/basic/select-popover.e2e.ts (hunk 1 above)
Comment (thetaPC) [reply core/src/components/select-popover/test/basic/select-popover.e2e.ts:88]: Good catch!

**Review by thetaPC — APPROVED** [review]
LGTM

**Review by brandyscarney — APPROVED** [review]
These changes are great! Nice catch for the `md` positioning too. I don't think we need to do this now, but I do think we should remove the last border on `ios`:

| Safari | Podcasts |
| ---| ---|
| IMG_3148 | IMG_3150 |

**Review by amandaejohnston — APPROVED** [review]

**Comment by Borongaj** [comment]
ion-select still shows warning with popover interface with Ionic 7.3.3.

**Comment by liamdebeasi** [comment]
@Borongaj Please file an issue with a minimal reproduction.

**Pull request merged**